src/figuresClasses.ts

```diff
@@ -1,11 +1,103 @@
-export interface Figure {}
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  lue = 'blue',
+}
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  protected sidesArray: number[] = [this.a, this.b, this.c];
+
+  protected longestSide: number = this.sidesArray.splice(
+    this.sidesArray.indexOf(Math.max(...this.sidesArray)),
+  )[0];
+
+  protected otherSidesSum: number = this.sidesArray.reduce(
+    (acc: number, sideLength: number) => acc + sideLength,
+    0,
+  );
+
+  protected semiperimeter: number = (this.longestSide + this.otherSidesSum) / 2;
 
-export class Triangle implements Figure {}
+  protected isSidesWrong: boolean = this.longestSide >= this.otherSidesSum;
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    readonly shape: Shape = Shape.Triangle,
+  ) {
+    if (this.sidesArray.some((side) => side <= 0)) {
+      throw new Error('length of some side is <= 0');
+    }
```

Comment:
you can create function like

function checkOnError(condition: boolean, errorMessage: string): void {
  if (condition) {
    throw new Error(errorMessage);
  }
}
It is needed because you have the same logic in three places.